Tests/SentryTests/Helper/SentryDeviceTests.mm

```diff
@@ -0,0 +1,188 @@
+    // Xcode 13.4.1 on an Apple Silicon MBP (armcknight 23 Sep 2022)
+    SENTRY_ASSERT_CONTAINS(arch, @"x86"); // Macs with Intel CPUs
+#    else
+    SENTRY_ASSERT_CONTAINS(arch, @"arm64"); // Macs with Apple Silicon
+#    endif
+#elif TARGET_OS_IOS
+#    if TARGET_OS_SIMULATOR
+#        if TARGET_CPU_ARM64
+    SENTRY_ASSERT_CONTAINS(arch, @"arm"); // iPhone simulator on M1 macs
+#        elif TARGET_CPU_X86_64
+    SENTRY_ASSERT_CONTAINS(arch, @"x86"); // iPhone simulator on Intel macs
+#        else
+    XCTFail(@"Unexpected CPU type on test host.");
+#        endif // TARGET_CPU_ARM64
+#    else
+    SENTRY_ASSERT_CONTAINS(arch, @"arm"); // Real iPads and iPhones
+#    endif
+#elif TARGET_OS_TV
+    // We must test this branch in tvOS-SwiftUITests since it must run on device, which SentryTests
+    // cannot.
+#    if TARGET_OS_SIMULATOR
+#        if TARGET_CPU_ARM64
+    SENTRY_ASSERT_CONTAINS(arch, @"arm"); // TV simulator on M1 macs
+#        elif TARGET_CPU_X86_64
+    SENTRY_ASSERT_CONTAINS(arch, @"x86"); // TV simulator on Intel macs
+#        else
+    XCTFail(@"Unexpected CPU type on test host.");
+#        endif // TARGET_CPU_ARM64
+#    else
+    SENTRY_ASSERT_CONTAINS(arch, @"arm"); // Real TVs
+#    endif
+#elif TARGET_OS_WATCH
+    // TODO: create a watch UI test target to test this branch as it cannot run on the watch
+    // simulator
+    SENTRY_ASSERT_CONTAINS(arch, @"arm"); // Real Watches
+#else
+    XCTFail(@"Unexpected device OS");
+#endif
+}
+
+- (void)testOSVersion
+{
+    XCTAssertNotEqual(getOSVersion().length, 0U);
+}
+
+- (void)testOSName
+{
+    const auto osName = getOSName();
+#if TARGET_OS_OSX
+    SENTRY_ASSERT_EQUAL(osName, @"macOS");
+#elif TARGET_OS_MACCATALYST
+    SENTRY_ASSERT_EQUAL(osName, @"iPadOS");
+#elif TARGET_OS_IOS
+    // We must test this branch in iOS-SwiftUITests since it must run on device, which SentryTests
+    // cannot.
+    if (UIDevice.currentDevice.userInterfaceIdiom == UIUserInterfaceIdiomPad) {
+        SENTRY_ASSERT_EQUAL(osName, @"iPadOS");
+    } else if (UIDevice.currentDevice.userInterfaceIdiom == UIUserInterfaceIdiomPhone) {
+        SENTRY_ASSERT_EQUAL(osName, @"iOS");
+    } else {
+        XCTFail(@"Unsupported iOS UI idiom.");
+    }
+#elif TARGET_OS_TV
+    // We must test this branch in tvOS-SwiftUITests since it must run on device, which SentryTests
+    // cannot.
+    SENTRY_ASSERT_EQUAL(osName, @"tvOS");
+#elif TARGET_OS_WATCH
+    // TODO: create a watch UI test target to test this branch as it cannot run on the watch
+    // simulator
+    SENTRY_ASSERT_EQUAL(osName, @"watchOS");
+#else
+    XCTFail(@"Unexpected device OS");
+#endif
+}
+
+- (void)testDeviceModel
+{
+    const auto modelName = getDeviceModel();
+    XCTAssertNotEqual(modelName.length, 0U);
+#if TARGET_OS_OSX || TARGET_OS_MACCATALYST
+#    if defined(TESTCI) || defined(TEST)
+    SENTRY_ASSERT_CONTAINS(modelName, @"VMware");
```

Comment:
You are right, but it would be pretty obvious what the issue is. I don't think we plan on changing CI very frequently, or that GHA will change the underlying platform very frequently, so this is hopefully a nonissue.